### PR TITLE
[IITO] Clients can share intended commonpoolcontribution

### PR DESCRIPTION
**Pull request by Eirikalb**
# Summary

Clients can share intended common pool contribution, lying will be possible so beware!

## Additional Information

Thanks Carla for help! 

## Test Plan

Added tests to server distribution function.

### CODE REVIEW

**Comment by QFSW** [comment]
# ⚠️Warning! This diff contains changes in protected folders

File: internal/common/baseclient/iito.go [hunk 1]
```diff
@@ -75,3 +76,33 @@ func (c *BaseClient) ReceivedGift(received shared.Resources, from shared.ClientI
 	// You can check your updated resources like this:
 	// myResources := c.ServerReadHandle.GetGameState().ClientInfo.Resources
 }
+
+// ShareIntendedContributionis called on each client to give them the
```
Comment (E-Mac27) [inline internal/common/baseclient/iito.go]:
```suggestion
// ShareIntendedContribution is called on each client to give them the 
```

File: internal/common/baseclient/iito.go [hunk 2]
```diff
@@ -75,3 +76,33 @@ func (c *BaseClient) ReceivedGift(received shared.Resources, from shared.ClientI
 	// You can check your updated resources like this:
 	// myResources := c.ServerReadHandle.GetGameState().ClientInfo.Resources
 }
+
+// ShareIntendedContributionis called on each client to give them the 
+// option of sharing how much they intend to contribute to the common pool. They must also
+// choose what islands they want to share the information with
```
Comment (E-Mac27) [inline internal/common/baseclient/iito.go]: Please add if this is Compulsory or Optional.

File: internal/common/baseclient/iito.go [hunk 3]
```diff
@@ -75,3 +76,33 @@ func (c *BaseClient) ReceivedGift(received shared.Resources, from shared.ClientI
...
+func (c *BaseClient) ShareIntendedContribution() shared.IntendedContribution {
+	
+	// For MVP, share this prediction with all islands since trust has not yet been implemented
+	trustedIslands := make([]shared.ClientID, len(RegisteredClients))
```
Comment (E-Mac27) [inline internal/common/baseclient/iito.go]: Use alive clients only. See https://github.com/SOMAS2020/SOMAS2020/blob/main/internal/common/baseclient/iito.go#L14

**Review by E-Mac27 — CHANGES_REQUESTED** [review]
:D This looks similar to the DisasterPrediction PR so nothing much to say. It's mainly baseclient stuff.

**Pull request merged**